cantarelli/app.js

```diff
@@ -0,0 +1,258 @@
+prestamos.push(new Prestamo(3, 2, 1, 3));
+
+app.get('/socios', function(req,res){
+    res.status(200).json(socios);    
+})
+
+/*para tener todos los libros*/
+app.get('/libros', function(req,res){
+    res.status(200).json(libros);      
+})
+
+app.get('/libros/:idlibro', function(req,res){      
+    res.status(200).json(disponibleLibro(req.params.idlibro));  
+})
+
+app.get('/prestamos', function(req,res){
+
+    res.status(200).json(prestamos);     
+})
+
+/*Obtener los libros prestados al socio con sus fechas de vencimiento.*/
+app.get('/socios/:idsocio/prestamos', function(req,res){
+    if(buscarIdSocio(req.params.idsocio) != 0){
+        res.status(400);
+        res.send("No se puede obtener los libros prestados al socio");
+    }else{
+        res.status(200).json(buscarPrestamos(req.params.idsocio));  
+    }   
+})
+
+/*actualiza*/
+app.put('/libros/:idlibro', function(req,res){
+    if(buscarIdLibro(req.params.idlibro) != 0){
+        res.status(400);
+        res.send("No se puede actualizar la cantidad de libros");
+    }else{
+        actualizarCantidad(req.params.idlibro,req.body.cantidad);
+        res.status(204).json(libros); 
+    }
+})
+
+/*coloca nuevo*/
+app.post('/socios', function(req,res){
+    socios.push(new Socio(req.body.id, req.body.nombre));    
+    res.json(socios);
+    res.status(201);
+})
+
+app.post('/libros', function(req,res){
+    libros.push(new Libro(req.body.id, req.body.nombre,req.body.cantidad));
+    res.status(201);    
+    res.json(libros);
+})
+
+/*Registrar un préstamo de un libro a un socio.*/
+app.post('/prestamos', function(req,res){
+    if(disponibleLibro(req.body.idLibro) > 0 && librosAdeudados(req.body.idSocio) == 0){
+        prestamos.push(new Prestamo(req.body.id, req.body.idLibro, req.body.idSocio, req.body.dias));
+        res.status(201);
+        res.json(prestamos);
+    }
+    else{
+        res.status(404);
+        res.send("No se puede realizar el prestamo");
+    }  
+})
+
+/* Un socio puede devolver un libro prestado*/
+app.delete('/prestamos/:idSocio/:idLibro',function(req,res){
+    if(buscarIdSocio(req.params.idSocio) != 0 || buscarIdLibro(req.params.idLibro) != 0){
+        res.status(400);
+        res.send("No se puede borrar el prestamo, no existe");
+    }else{
+        if(borrarPrestamo(req.params.idSocio, req.params.idLibro)){
+            res.status(204).send("Se elimino el prestamo");
+        }else{
+            res.status(400).send("No se puede eliminar el prestamo");
+        }              
+    }   
+})
+
+app.delete('/libros/:id',function(req,res){
+    if(buscarIdLibro(req.params.id) != 0){
+        res.status(400).send("No se puede eliminar el libro");
+    } else{
+        if(borrarLibro(req.params.id)){            
+            res.status(204).send("Se elimino el libro");
+        }else{
+            res.status(400).send("No se puede eliminar el libro");
+        }        
+    }
+})
+
+var server = app.listen(8080, '127.0.0.1', function() {
```

Comment:
The host is set to '127.0.0.1' or 'localhost' by default, you can avoid that paramether